app/helpers/prominence_helper.rb

```diff
@@ -0,0 +1,129 @@
+##
+# Methods to render consistently object's prominence and prominence_reason
+# depending on the context (prominence value, current user, format)
+#
+module ProminenceHelper
+  def conceled_prominence?(object)
+    %w{hidden requester_only}.include?(object.prominence)
+  end
+
+  def render_prominence(object, format: :html)
+    notice = prominence_notice(object)
+
+    return unless notice
+    return notice + "\n" * 2 if format == :text
+
+    [notice, prominence_sign_in(object), prominence_contact_us].
+      join(' ').squish.html_safe
+  end
+
+  private
+
+  def prominence_notice(object)
+    case object
+    when InfoRequest
+      case object.prominence
+      when 'hidden'
+        prominence_for_hidden_request(object)
+      when 'requester_only'
+        prominence_for_requester_only_request(object)
+      end
+
+    when IncomingMessage, OutgoingMessage
+      case object.prominence
+      when 'hidden'
+        prominence_for_hidden_message(object)
+      when 'requester_only'
+        prominence_for_requester_only_message(object)
+      end
+    end
```

Comment:
I do think this could do with an `else` that raises an `ArgumentError` if an unhandled prominenceable is provided. Reason being is – in a few years time when I've forgot about this and we add a new object that's prominenceable – just expect to call `render_prominence(my_new_prominenceable)` and it work. I'd currently get no output, which would be easy to forget / miss.

This _would_ mean we'd also need to handle normal and backpage prominence. This looks like the easiest  place to do that for now?

```diff
   def render_prominence(object, format: :html)
+    return nil unless conceled_prominence?(object)
```